src/Core/InstanceConfigTrait.php

```diff
@@ -40,6 +41,21 @@ trait InstanceConfigTrait
      */
     protected $_configInitialized = false;
 
+    /**
+     * Gets the whole config array.
+     *
+     * @return array
+     */
+    public function config(): array
```

Comment:
Not too sure about the method name. Given that in the past getConfig() was named config() it can be confusing.

Perhaps name this method allConfig() or getAllConfig().

Reply:
Is `getEntireConfig` against cakephp naming conventions? I think that might read easier.

Reply:
It's not against convention but we don't have any other method with "Entire" in its name. `getAllConfig()` would be more consistent naming.

Reply:
config() is also the name of the deprecated method in 3.x. Someone upgrading could have a subtle failure if they didn't address all their deprecations.